controllers/extensionRequests.js

```diff
@@ -0,0 +1,233 @@
+const extensionRequestsQuery = require("../models/extensionRequests");
+const { addLog } = require("../models/logs");
+const tasks = require("../models/tasks");
+const { getUsername } = require("../utils/users");
+const { EXTENSION_REQUEST_STATUS } = require("../constants/extensionRequests");
+
+/**
+ * Create ETA extension Request
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const createTaskExtensionRequest = async (req, res) => {
+  try {
+    const extensionBody = req.body;
+
+    if (req.userData.id !== extensionBody.assignee && !req.userData.roles?.super_user) {
+      return res.boom.forbidden("Only Super User can create an extension request for this task.");
+    }
+
+    const assigneeUsername = await getUsername(extensionBody.assignee);
+    const { taskData: task } = await tasks.fetchTask(extensionBody.taskId);
+    if (!task) {
+      return res.boom.badRequest("Task with this id or taskid doesn't exist.");
+    }
+    if (task.assignee !== assigneeUsername) {
+      return res.boom.badRequest("This task is assigned to some different user");
+    }
+    if (task.endsOn >= extensionBody.newEndsOn) {
+      return res.boom.badRequest("The value for newEndsOn should be greater than the previous ETA");
+    }
+    if (extensionBody.oldEndsOn !== task.endsOn) {
+      extensionBody.oldEndsOn = task.endsOn;
+    }
```

Comment:
NIT: this line confuses a bit above this, you are using early return if the value is less, why can't I check if the value is more than oldEndsOn here?

This is handled by the early return but was a bit confusing to me.

Reply:
great point, let me recheck this. 👍

Reply:
so basically the reason I do this, is to make sure even if the user gives the wrong `oldEndsOn` I will just replace it with the current one from the main task object